src/main/java/dk/alexandra/fresco/lib/helper/builder/AbstractProtocolBuilder.java

```diff
@@ -146,7 +177,19 @@ public ProtocolProducer getProtocol() {
 	
 	@Override
 	public void reset() {
+		if(parent != null){
+			parent.reset();
+			return;
+		}
 		curpp = new SequentialProtocolProducer();
 		producerStack = new Stack<AppendableProtocolProducer>();
 	}
+	
+	/**
+	 * Sets the parent of this builder meaning that the internal stack of the builder will be the parents. 
+	 * @param parent
+	 */
+	protected void setParentBuilder(AbstractProtocolBuilder parent){
```

Comment:
You would only have a single structural builder. It could be the OmniBuilder, but as we already talked about I am not sure I like the idea of the OmniBuilder to begin with ... the point is that it is really messy that you can control scope using the sub builders provided by the OmniBuilder.